### PR TITLE
feat: add SourceCodeEditor form control #714

### PR DESCRIPTION
**Pull request by ilya-trifonov**
affects: @haulmont/jmix-react-antd, test-puppeteer

### CODE REVIEW

File: example-react-app/src/app/custom-form-controls/CustomFormControls.tsx [hunk 1]
```diff
@@ -122,6 +126,17 @@ const CustomFormControls = () => {
...
         />
+        <SourceCodeEditor
+          entityName="scr_Car"
+          propertyName="createdBy"
+          formItemProps={{
+            style: { marginBottom: "12px" }
+          }}
+          mode="javascript"
+          theme="github"
+          name="UNIQUE_ID_OF_DIV"
```
Comment (vyacheslav-pushkin) [inline example-react-app/src/app/custom-form-controls/CustomFormControls.tsx:137]: Can we put it inside `SourceCodeField` and default it to `propertyName`?

File: example-react-app/src/app/custom-form-controls/CustomFormControls.tsx [hunk 2]
```diff
@@ -122,6 +126,17 @@ const CustomFormControls = () => {
...
+          editorProps={{ $blockScrolling: true }}
```
Comment (vyacheslav-pushkin) [inline example-react-app/src/app/custom-form-controls/CustomFormControls.tsx:138]: Maybe put inside `SourceCodeField` too.

File: scripts/custom-screens/CustomFormControls.tsx [hunk 3]
```diff
@@ -14,11 +14,14 @@ import {
...
   MaskedField,
-  SliderField
+  SliderField,
+  SourceCodeEditor
 } from "@haulmont/jmix-react-antd";
 import { FormattedMessage, useIntl } from "react-intl";
 import { useForm } from "antd/es/form/Form";
 import dayjs from "dayjs";
+import "ace-builds/src-noconflict/mode-javascript";
+import "ace-builds/src-noconflict/theme-github";
```
Comment (vyacheslav-pushkin) [inline scripts/custom-screens/CustomFormControls.tsx:24]: Can we avoid having to make these imports manually?

File: example-react-app/src/app/custom-form-controls/CustomFormControls.tsx [hunk 4]
```diff
@@ -122,6 +126,17 @@ const CustomFormControls = () => {
...
           }}
         />
+        <SourceCodeEditor
+          entityName="scr_Car"
+          propertyName="createdBy"
+          formItemProps={{
+            style: { marginBottom: "12px" }
+          }}
+          mode="javascript"
+          theme="github"
```
Comment (vyacheslav-pushkin) [inline example-react-app/src/app/custom-form-controls/CustomFormControls.tsx:136]: As discussed

**Review by vyacheslav-pushkin — CHANGES_REQUESTED** [review]

File: packages/jmix-react-antd/package.json [hunk 5]
```diff
@@ -22,6 +22,7 @@
     "mobx": "^6.3.3",
     "mobx-react": "^7.2.0",
     "react": "^17.0.1",
+    "react-ace": "^9.5.0",
```
Comment (vyacheslav-pushkin) [inline packages/jmix-react-antd/package.json:25]: It should be dependency, not peer dependency.

**Review by vyacheslav-pushkin — CHANGES_REQUESTED** [review]

File: example-react-app/src/app/custom-form-controls/CustomFormControls.tsx [hunk 6]
```diff
@@ -14,11 +14,14 @@ import {
   TimeField,
   TextArea,
   MaskedField,
-  SliderField
+  SliderField,
+  SourceCodeEditor
 } from "@haulmont/jmix-react-antd";
```
Comment (vyacheslav-pushkin) [inline example-react-app/src/app/custom-form-controls/CustomFormControls.tsx:19]: example-app doesn't start:

```
Failed to compile.

./src/app/custom-form-controls/CustomFormControls.tsx
Attempted import error: 'SourceCodeEditor' is not exported from '@haulmont/jmix-react-antd'.
```

**Review by vyacheslav-pushkin — CHANGES_REQUESTED** [review]

File: scripts/custom-screens/CustomFormControls.tsx (hunk 3 above)
Comment (vyacheslav-pushkin) [reply scripts/custom-screens/CustomFormControls.tsx]: How is this resolved? I still see the import.

Comment (vyacheslav-pushkin) [reply scripts/custom-screens/CustomFormControls.tsx]: What are these imports for?

Comment (vyacheslav-pushkin) [reply scripts/custom-screens/CustomFormControls.tsx]: OK, I see, it seems to be the way `react-ace` works.

Please add information to QA Notes about what imports needs to be added in order for the component to work.

Comment (ilya-trifonov) [reply scripts/custom-screens/CustomFormControls.tsx]: Yes, without these scripts the code editor will not work. They are needed for syntax highlighting and editor theme.
User sees only the script for highlighting the language now.

**Review by vyacheslav-pushkin — APPROVED** [review]

**Review by vyacheslav-pushkin — APPROVED** [review]

**Comment by vyacheslav-pushkin** [comment]
@ilya-trifonov Please resolve conflicts and merge

**Pull request merged**